src/context.rs

```diff
@@ -323,7 +404,33 @@ impl<'a, 'b> ExtJsonBody for BodyReader<'a, 'b> {
 impl<'a, 'b> Read for BodyReader<'a, 'b> {
     ///Read the request body.
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        self.request.read(buf)
+        self.reader.read(buf)
+    }
+}
+
+///A specialized request representation for the multipart interface.
+#[cfg(feature = "multipart")]
+pub struct MultipartRequest<'r, 'a: 'r, 'b: 'a> {
+    boundary: &'r str,
+    reader: &'r mut HttpReader<&'a mut BufReader<&'b mut NetworkStream>>
+}
+
+#[cfg(feature = "multipart")]
+impl<'r, 'a, 'b> HttpRequest for MultipartRequest<'r, 'a, 'b> {
+    fn is_multipart(&self) -> bool {
+        true
```

Comment:
Haha, yeah, the presence of a boundary string should be enough :smile: I can't imagine that it would be too hard to find while checking if it's a valid multipart response.

I haven't merged it yet, so I'm completely ok with refactoring. It's better to fix it as soon as possible :+1: